src/root/components/FinalReportExport/FinalReportPdfDocument/index.tsx

```diff
@@ -0,0 +1,828 @@
+              </Text>
+            </View>
+            <View style={[pdfStyles.piHeaderCell, pdfStyles.fontWeightNormalAndSmall]}>
+              <Text style={pdfStyles.fontWeightNormalAndSmall}>
+                {indicator?.actual}
+              </Text>
+            </View>
+          </View>
+        ))
+      }
+      <View style={pdfStyles.piRow}>
+        <Text style={[pdfStyles.piBorderCell, pdfStyles.fontWeightBoldAndLarge]}>
+          {strings.finalReportPlannedInterventionNarrativeAchievement}
+        </Text>
+      </View>
+      {data?.challenges && (
+        <View style={pdfStyles.piRow}>
+          <Text style={pdfStyles.piBorderCell}>
+            {data?.narrative_description_of_achievements}
+          </Text>
+        </View>
+      )}
+      <View style={pdfStyles.piRow}>
+        <Text style={[pdfStyles.piBorderCell, pdfStyles.fontWeightBoldAndLarge]}>
+          {strings.finalReportPlannedInterventionChallenges}
+        </Text>
+      </View>
+      {data?.challenges && (
+        <View style={pdfStyles.piRow}>
+          <Text style={pdfStyles.piBorderCell}>
+            {data?.challenges}
+          </Text>
+        </View>
+      )}
+      <View style={pdfStyles.piRow}>
+        <Text style={[pdfStyles.piBorderCell, pdfStyles.fontWeightBoldAndLarge]}>
+          {strings.finalReportPlannedInterventionLessonsLearnt}
+        </Text>
+      </View>
+      {data?.lessons_learnt && (
+        <View style={pdfStyles.piRow}>
+          <Text style={pdfStyles.piBorderCell}>
+            {data?.lessons_learnt}
+          </Text>
+        </View>
+      )}
+    </View>
+  );
+}
+
+interface DrefOptions {
+  disaster_category: NumericKeyValuePair[];
+  needs_identified: StringKeyValuePair[];
+  planned_interventions: StringKeyValuePair[];
+  status: NumericKeyValuePair[];
+  type_of_onset: NumericKeyValuePair[];
+  users: {
+    id: number;
+    first_name: string;
+    last_name: string;
+    email: string;
+    username: string;
+  }[];
+}
+
+interface Props {
+  finalReportResponse: DrefFinalReportApiFields;
+  drefOptions: DrefOptions;
+  strings: Strings;
+}
+
+const logoUrl = '/assets/graphics/layout/go-logo-2020.png';
+
+function FinalReportPdfDocument(props: Props) {
+  const {
+    finalReportResponse,
+    drefOptions,
+    strings,
+  } = props;
+
+  const niMap = useMemo(() => (
+    listToMap(
+      drefOptions?.planned_interventions,
+      intervention => intervention.key,
+      intervention => intervention.value,
+    )
+  ), [drefOptions]);
+
+  const piMap = useMemo(() => (
+    listToMap(
+      drefOptions?.needs_identified,
+      need => need.key,
+      need => need.value,
+    )
+  ), [drefOptions]);
+
+  const affectedAreas = useMemo(() => {
+    const districts = finalReportResponse?.country_district.map(d => d['district_details']);
+    const areas = districts.map(districtsList => {
+      return districtsList.map((districtItem) => districtItem.name).join(', ');
+    }).join(', ');
```

Comment:
Let's only join once if all of them are separate by `,`
Let's flat the array and join by `,`
Also, we may need to check for unique values.